nixos/modules/services/network-filesystems/dave.nix

```diff
@@ -0,0 +1,143 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.dave;
+  settingsFormat = pkgs.formats.yaml { };
+in
+{
+  options.services.dave = {
+    enable = mkEnableOption "dave WebDAV server";
+
+    user = mkOption {
+      type = types.str;
+      default = "dave-webdav";
+      description = "User account under which dave runs.";
+    };
+
+    group = mkOption {
+      type = types.str;
+      default = "dave-webdav";
+      description = "Group under which dave runs.";
+    };
+
+    openFirewall = mkOption {
+      type = types.bool;
+      default = true;
+      description = ''
+        Whether to automatically open the specified ports in the firewall.
+      '';
+    };
+
+    settings = mkOption {
+      type = types.submodule {
+        freeformType = settingsFormat.type;
+
+        options = {
+          address = mkOption {
+            type = types.str;
+            default = "127.0.0.1";
+            description = "The bind address.";
+          };
+          port = mkOption {
+            type = types.port;
+            default = 8000;
+            description = "The listening port.";
+          };
+          dir = mkOption {
+            type = types.str;
+            description = "The provided base dir.";
+          };
+        };
+      };
+      default = { };
+      example = {
+        address = "127.0.0.1";
+        port = 8000;
+        dir = "/home/webdav";
+        prefix = "/webdav";
+        users = {
+          user = {
+            password =
+              "$2a$10$yITzSSNJZAdDZs8iVBQzkuZCzZ49PyjTiPIrmBUKUpB0pwX7eySvW";
+            subdir = "/user";
+          };
+          admin = {
+            password =
+              "$2a$10$DaWhagZaxWnWAOXY0a55.eaYccgtMOL3lGlqI3spqIBGyM0MD.EN6";
+          };
+        };
+      };
+      description = ''
+        Configuration for dave. See
+        <link xlink:href="https://github.com/micromata/dave/blob/master/Readme.md"/>
+        for supported values.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    users = {
+      users = mkIf (cfg.user == "dave-webdav") {
+        dave-webdav = {
+          description = "dave webdav server user";
+          isSystemUser = true;
+          group = cfg.group;
+        };
+      };
+      groups = mkIf (cfg.group == "dave-webdav") { dave-webdav = { }; };
+    };
+
+    systemd =
+      let workDir = "/run/dave";
+      in
+      {
+        services.dave = {
+          description = "dave - The simple webdav server";
+          after = [ "syslog.target" "network.target" "sysinit.target" ];
+          wantedBy = [ "multi-user.target" ];
+          serviceConfig = {
+            ExecStart = "${pkgs.dave}/bin/dave";
+            WorkingDirectory = workDir;
+
+            User = cfg.user;
+            Group = cfg.group;
+
+            CapabilityBoundingSet = "";
```

Comment:
Are these sandboxing options from trial & error? Or did you find them suggested somewhere?

For bonus points... did you want to contribute a `systemd` service to the upstream repo?

Reply:
Yes trial and error (the only suggestions come from systemd.exec(5)). I would be happy to contribute a service upstream, but it will be easier once/if some kind of `--config` argument is added.